packages/turf-helpers/geojson.d.ts

```diff
@@ -0,0 +1,238 @@
+                            "MultiPoint" |
+                            "MultiLineString" |
+                            "MultiPolygon";
+
+export type CollectionTypes = "FeatureCollection" | "GeometryCollection";
+
+/**
+ * Types
+ *
+ * https://tools.ietf.org/html/rfc7946#section-1.4
+ * The value values for the "type" property of GeoJSON Objects.
+ */
+export type Types = "Feature" | GeometryTypes | CollectionTypes;
+
+/**
+ * Bounding box
+ * 
+ * https://tools.ietf.org/html/rfc7946#section-5
+ * A GeoJSON object MAY have a member named "bbox" to include information on the coordinate range for its Geometries, Features, or FeatureCollections.
+ * The value of the bbox member MUST be an array of length 2*n where n is the number of dimensions represented in the contained geometries,
+ * with all axes of the most southwesterly point followed by all axes of the more northeasterly point.
+ * The axes order of a bbox follows the axes order of geometries.
+ */
+export type BBox2d = [number, number, number, number];
+export type BBox3d = [number, number, number, number, number, number];
+export type BBox = BBox2d | BBox3d;
+
+/**
+ * Id
+ *
+ * https://tools.ietf.org/html/rfc7946#section-3.2
+ * If a Feature has a commonly used identifier, that identifier SHOULD be included as a member of
+ * the Feature object with the name "id", and the value of this member is either a JSON string or number.
+ */
+export type Id = string | number;
+
+/**
+ * Position
+ * 
+ * https://tools.ietf.org/html/rfc7946#section-3.1.1
+ * Array should contain between two and three elements.
+ * The previous GeoJSON specification allowed more elements (e.g., which could be used to represent M values),
+ * but the current specification only allows X, Y, and (optionally) Z to be defined.
+ */
+export type Position = number[]; // [number, number] | [number, number, number];
+
+/**
+ * Properties
+ *
+ * https://tools.ietf.org/html/rfc7946#section-3.2
+ * A Feature object has a member with the name "properties".
+ * The value of the properties member is an object (any JSON object or a JSON null value).
+ */
+export type Properties = { [name: string]: any; } | null;
+
+/**
+ * Geometries
+ */
+export type Geometries = Point |
+                         LineString |
+                         Polygon |
+                         MultiPoint |
+                         MultiLineString |
+                         MultiPolygon;
+
+/**
+ * GeoJSON Object
+ *
+ * https://tools.ietf.org/html/rfc7946#section-3
+ * The GeoJSON specification also allows [foreign members](https://tools.ietf.org/html/rfc7946#section-6.1)
+ * Developers should use "&" type in TypeScript or extend the interface to add these foreign members.
+ */
+export interface GeoJSONObject {
+    // Don't include foreign members directly into this type def.
+    // in order to preserve type safety.
+    // [key: string]: any;
+    /**
+     * Specifies the type of GeoJSON object.
+     */
+    type: string;
```

Comment:
@zakjan To solve your issue https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21590#issuecomment-348153458 I actually converted `GeoJSONObject` as `type: string`.

Maybe you should suggest a change to the new `@types/geojson` definition.